File: tests/manual/texmark_raylib/sprites.py
# CPython texmark script (same shape as sprites.zen).
import native

class Sprite:
    __slots__ = ("tex", "x", "y", "vx", "vy")

    def __init__(self, tex, x, y, vx, vy):
        self.tex = tex
        self.x = x
        self.y = y
        self.vx = vx
        self.vy = vy

    def update(self, dt):
        self.x = self.x + self.vx * dt
        self.y = self.y + self.vy * dt
        if self.x > max_x or self.x < 0:
            self.vx = -self.vx
        if self.y > max_y or self.y < 0:
            self.vy = -self.vy
        self.tex.draw(self.x, self.y)

wabbit = native.Texture("wabbit_alpha.png")
sprites = []
max_x = native.screen_width() - wabbit.width()
max_y = native.screen_height() - wabbit.height()

def add_sprites(n, x, y):
    rand = native.rand
    for _ in range(n):
        if x < 0:
            sprites.append(Sprite(wabbit, rand(0, max_x), rand(0, max_y), rand(-250, 250), rand(-250, 250)))
        else:
            sprites.append(Sprite(wabbit, x, y, rand(-250, 250), rand(-250, 250)))

def update_all(dt):
    for s in sprites:
        s.update(dt)

File: tests/manual/texmark_raylib/sprites.lua
-- Lua 5.4 texmark script (same shape as sprites.zen).
local Sprite = {}
Sprite.__index = Sprite

function Sprite.new(tex, x, y, vx, vy)
    return setmetatable({ tex = tex, x = x, y = y, vx = vx, vy = vy }, Sprite)
end

function Sprite:update(dt)
    self.x = self.x + self.vx * dt
    self.y = self.y + self.vy * dt
    if self.x > max_x or self.x < 0 then
        self.vx = -self.vx
    end
    if self.y > max_y or self.y < 0 then
        self.vy = -self.vy
    end
    self.tex:draw(self.x, self.y)
end

wabbit = Texture("wabbit_alpha.png")
sprites = {}
max_x = screen_width() - wabbit:width()
max_y = screen_height() - wabbit:height()

function add_sprites(n, x, y)
    local list = sprites
    for i = 1, n do
        if x < 0 then
            list[#list + 1] = Sprite.new(wabbit, rand(0, max_x), rand(0, max_y), rand(-250, 250), rand(-250, 250))
        else
            list[#list + 1] = Sprite.new(wabbit, x, y, rand(-250, 250), rand(-250, 250))
        end
    end
end

function update_all(dt)
    local list = sprites
    for i = 1, #list do
        list[i]:update(dt)
    end
end
